velox/connectors/hive/SplitReader.cpp

```diff
@@ -91,7 +92,19 @@ void SplitReader::prepareSplit(
   VELOX_CHECK_NE(
       baseReaderOpts_.getFileFormat(), dwio::common::FileFormat::UNKNOWN);
 
-  auto fileHandle = fileHandleFactory_->generate(hiveSplit_->filePath).second;
+  std::shared_ptr<FileHandle> fileHandle;
+  try {
+    fileHandle = fileHandleFactory_->generate(hiveSplit_->filePath).second;
```

Comment:
Would you also document 'generate' API to explain that it should throw a specific error code to indicate file-not-found?

Reply:
I plan to add doc for this in the follow up pr, sth like: https://github.com/facebookincubator/velox/pull/8645/commits/c195d903fedfeaded65ee0a25b0874bc1321973d#diff-b2b8b019f80c44db000432c6c90bfffeb215c772b15908246f736ddd30011d1eR84. Please let me know if other docs are needed, thanks.